### PR TITLE
luarocks-nix: bump

### LINKED ISSUE #163134: outdated luaPackages.luv
###### Checklist

I hit this when developing on neovim https://github.com/neovim/neovim/issues/17562 . which made me realize realized I had broken the luarocks updater (I now have a fix for it). Also the uris keep changing when updating the lua set, I think we could use mirror:// more there

- [ ] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [ ] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name:

current version:
desired version:

###### Notify maintainers

maintainers:

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by teto**
I wanted to hack on neovim but got held up by several things. Trying to address those here while improving lua ecosystem at the same time.

Related: 
https://github.com/NixOS/nixpkgs/pull/80528 
https://github.com/NixOS/nixpkgs/pull/141718

Fixes https://github.com/NixOS/nixpkgs/issues/163134 .

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teto** [comment]
fixes https://github.com/NixOS/nixpkgs/issues/163134

**Comment by teto** [comment]
@doronbehar you should be happy this makes the libluv packaging a bit clearer IMHO.

**Comment by doronbehar** [comment]
Thanks a lot @teto for the hard work! Well done. I pushed a few additions to the neovim expression.

**Comment by doronbehar** [comment]
# NOTE: DO not merge now, the last commit should be squashed with the one before it, I'm using ofborg to test darwin builds.

File: pkgs/applications/editors/neovim/default.nix [hunk 1]
```diff
@@ -97,12 +93,12 @@ in
...
-      "-DGPERF_PRG=${gperf}/bin/gperf"
-      "-DLUA_PRG=${neovimLuaEnv.interpreter}"
-      "-DLIBLUV_LIBRARY=${luvpath}"
+      # Don't use downloaded dependencies. At the end of the configurePhase one
+      # can spot that cmake says this option was "not used by the project".
+      # That's because all dependencies were found and
+      # third-party/CMakeLists.txt is not read at all.
       "-DUSE_BUNDLED=OFF"
     ]
-    ++ optional doCheck "-DBUSTED_PRG=${neovimLuaEnv}/bin/busted"
```
Comment (teto) [inline pkgs/applications/editors/neovim/default.nix:105]: have you checked `make functionaltest` work without this ?

Comment (doronbehar) [reply pkgs/applications/editors/neovim/default.nix:105]: Apparently, the `checkPhase` fails with and without this line:

```
neovim-unwrapped> [ RUN      ] user config init loads init.lua from XDG config home by default: -- Tests exited non-zero: Process terminated due to timeout
neovim-unwrapped> -- Output to stderr:
neovim-unwrapped> sh: invalid number '--'
neovim-unwrapped> sh: invalid number '--'
neovim-unwrapped> sh: invalid number '--'
neovim-unwrapped> CMake Error at /build/source/cmake/RunTests.cmake:98 (message):
neovim-unwrapped>   functional tests failed with error: Process terminated due to timeout
```

Comment (teto) [reply pkgs/applications/editors/neovim/default.nix:105]: I've tested this PR without your commit and I could run `make functionaltest` (though some tests fail but it has always been the case). Could you check it was the case for you ? and if yes revert the changes that broke it. I use make functionaltest locally to develop on neovim. Tests are too flaky to enable it for everyone but it's important to keep the ability to run them.

Comment (doronbehar) [reply pkgs/applications/editors/neovim/default.nix:105]: Tests fail for me even on branch master :/

**Comment by teto** [comment]
I've just checked with and without your commits, the functional tests start which is what I wanted to make sure. 

```
$ nix develop ./contrib --override-input nixpkgs /home/teto/nixpkgs2
$ rm -rf build/ && cmakeConfigurePhase
$ buildPhase

(ins)❯ make runtime
[  0%] Built target nvim_runtime_deps
[  0%] Built target update_version_stamp
Consolidate compiler generated dependencies of target nvim
[100%] Built target nvim
[100%] Generating syntax/vim/generated.vim
[100%] Generating doc/tags
[100%] Generating pack/dist/opt/matchit/doc/tags
[100%] Generating pack/dist/opt/vimball/doc/tags
[100%] Built target runtime
neovim/build on  treesitter-show-ft via △ v3.22.2 via ❄️  IMPURE (neovim-unwrapped-master) took 4s 
(ins)➜ make functionaltest
Consolidate compiler generated dependencies of target streams-test
[  0%] Built target streams-test
[  0%] Built target update_version_stamp
[100%] Built target nvim
Consolidate compiler generated dependencies of target tty-test
[100%] Built target tty-test
Consolidate compiler generated dependencies of target shell-test
[100%] Built target shell-test
Consolidate compiler generated dependencies of target printargs-test
[100%] Built target printargs-test
Consolidate compiler generated dependencies of target printenv-test
[100%] Built target printenv-test
[----------] Global test environment setup.
[----------] Running tests from test/functional/api/autocmd_spec.lua
[ RUN      ] autocmd api nvim_create_autocmd does not allow "command" and "callback" in the same autocmd: 14.76 ms OK
[ RUN      ] autocmd api nvim_create_autocmd doesnt leak when you use ++once: 14.48 ms OK
[ RUN      ] autocmd api nvim_create_autocmd allows passing buffer by key: 16.43 ms OK
[ RUN      ] autocmd api nvim_create_autocmd does not allow passing buffer and patterns: 13.49 ms OK
[ RUN      ] autocmd api nvim_create_autocmd does not allow passing invalid buffers: 13.86 ms OK
[ RUN      ] autocmd api nvim_create_autocmd errors on non-functions for cb: 13.44 ms OK
[ RUN      ] autocmd api nvim_create_autocmd allow passing pattern and <buffer> in same pattern: 13.29 ms OK

then some of them fail but we dont care
```

As far as I am concerned it's ok to merge. If a darwin user could confirm it works, we could merge.

**Comment by teto** [comment]
```
nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /home/ubuntu/.cache/nixpkgs-review/pr-166162/build.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/166162

10 packages marked as broken and skipped:
lua51Packages.digestif lua52Packages.digestif lua52Packages.gitsigns-nvim lua53Packages.gitsigns-nvim lua53Packages.ljsyscall lua53Packages.luabitop luaPackages.digestif luaPackages.gitsigns-nvim luajitPackages.digestif luajitPackages.luaffi

390 packages built:
abuild alpine-make-vm-image ani-cli anki ankisyncd apk-tools awesome celluloid curseradio d-seams gnvim haruna hydrus jellyfin-media-player jellyfin-mpv-shim jftui knot-resolver lua51Packages.alt-getopt lua51Packages.argparse lua51Packages.basexx lua51Packages.binaryheap lua51Packages.bit32 lua51Packages.busted lua51Packages.cassowary lua51Packages.cjson lua51Packages.compat53 lua51Packages.cosmo lua51Packages.coxpcall lua51Packages.cqueues lua51Packages.cyrussasl lua51Packages.dkjson lua51Packages.fifo lua51Packages.gitsigns-nvim lua51Packages.http lua51Packages.inspect lua51Packages.ldbus lua51Packages.ldoc lua51Packages.lgi luaPackages.libluv lua51Packages.linenoise lua51Packages.ljsyscall lua51Packages.lpeg lua51Packages.lpeg_patterns lua51Packages.lpeglabel lua51Packages.lpty lua51Packages.lrexlib-gnu lua51Packages.lrexlib-pcre lua51Packages.lrexlib-posix lua51Packages.lua-cmsgpack lua51Packages.lua-iconv lua51Packages.lua-lsp lua51Packages.lua-messagepack lua51Packages.lua-resty-http lua51Packages.lua-resty-jwt lua51Packages.lua-resty-openidc lua51Packages.lua-resty-openssl lua51Packages.lua-resty-session lua51Packages.lua-term lua51Packages.lua-toml lua51Packages.lua-yajl lua51Packages.lua-zlib lua51Packages.lua_cliargs lua51Packages.luabitop lua51Packages.luacheck lua51Packages.luacov lua51Packages.luadbi lua51Packages.luadbi-mysql lua51Packages.luadbi-postgresql lua51Packages.luadbi-sqlite3 lua51Packages.luaepnf lua51Packages.luaevent lua51Packages.luaexpat lua51Packages.luaffi lua51Packages.luafilesystem lua51Packages.lualogging lua51Packages.luaossl lua51Packages.luaposix lua51Packages.luarepl lua51Packages.luarocks lua51Packages.luarocks-nix lua51Packages.luasec lua51Packages.luasocket lua51Packages.luasql-sqlite3 lua51Packages.luassert lua51Packages.luasystem lua51Packages.luautf8 lua51Packages.luazip lua51Packages.luuid lua51Packages.luv lua51Packages.lyaml lua51Packages.markdown lua51Packages.mediator_lua lua51Packages.moonscript lua51Packages.mpack lua51Packages.nvim-client lua51Packages.penlight lua51Packages.plenary-nvim lua51Packages.rapidjson lua51Packages.readline lua51Packages.say lua51Packages.std-_debug lua51Packages.std-normalize lua51Packages.stdlib lua51Packages.vstruct luaPackages.alt-getopt luaPackages.argparse luaPackages.basexx luaPackages.binaryheap luaPackages.bit32 luaPackages.busted luaPackages.cassowary luaPackages.cjson luaPackages.compat53 luaPackages.cosmo luaPackages.coxpcall luaPackages.cqueues luaPackages.cyrussasl luaPackages.dkjson luaPackages.fifo luaPackages.http luaPackages.inspect luaPackages.ldbus luaPackages.ldoc luaPackages.lgi luaPackages.linenoise luaPackages.ljsyscall luaPackages.lpeg luaPackages.lpeg_patterns luaPackages.lpeglabel luaPackages.lpty luaPackages.lrexlib-gnu luaPackages.lrexlib-pcre luaPackages.lrexlib-posix luaPackages.lua-cmsgpack luaPackages.lua-iconv luaPackages.lua-lsp luaPackages.lua-messagepack luaPackages.lua-resty-http luaPackages.lua-resty-jwt luaPackages.lua-resty-openidc luaPackages.lua-resty-openssl luaPackages.lua-resty-session luaPackages.lua-term luaPackages.lua-toml luaPackages.lua-yajl luaPackages.lua-zlib luaPackages.lua_cliargs luaPackages.luabitop luaPackages.luacheck luaPackages.luacov luaPackages.luadbi luaPackages.luadbi-mysql luaPackages.luadbi-postgresql luaPackages.luadbi-sqlite3 luaPackages.luaepnf luaPackages.luaevent luaPackages.luaexpat luaPackages.luaffi luaPackages.luafilesystem luaPackages.lualogging luaPackages.luaossl luaPackages.luaposix luaPackages.luarepl luarocks luarocks-nix luaPackages.luasec luaPackages.luasocket luaPackages.luasql-sqlite3 luaPackages.luassert luaPackages.luasystem luaPackages.luautf8 luaPackages.luazip luaPackages.luuid luaPackages.luv luaPackages.lyaml luaPackages.markdown luaPackages.mediator_lua luaPackages.moonscript luaPackages.mpack luaPackages.nvim-client luaPackages.penlight luaPackages.plenary-nvim luaPackages.rapidjson luaPackages.readline luaPackages.say luaPackages.std-_debug luaPackages.std-normalize luaPackages.stdlib luaPackages.vstruct lua53Packages.alt-getopt lua53Packages.argparse lua53Packages.basexx lua53Packages.binaryheap lua53Packages.bit32 lua53Packages.busted lua53Packages.cassowary lua53Packages.cjson lua53Packages.compat53 lua53Packages.cosmo lua53Packages.coxpcall lua53Packages.cqueues lua53Packages.cyrussasl lua53Packages.digestif lua53Packages.dkjson lua53Packages.fifo lua53Packages.http lua53Packages.inspect lua53Packages.ldbus lua53Packages.ldoc lua53Packages.lgi lua53Packages.linenoise lua53Packages.lpeg lua53Packages.lpeg_patterns lua53Packages.lpeglabel lua53Packages.lpty lua53Packages.lrexlib-gnu lua53Packages.lrexlib-pcre lua53Packages.lrexlib-posix lua53Packages.lua-cmsgpack lua53Packages.lua-iconv lua53Packages.lua-lsp lua53Packages.lua-messagepack lua53Packages.lua-resty-http lua53Packages.lua-resty-jwt lua53Packages.lua-resty-openidc lua53Packages.lua-resty-openssl lua53Packages.lua-resty-session lua53Packages.lua-term lua53Packages.lua-toml lua53Packages.lua-yajl lua53Packages.lua-zlib lua53Packages.lua_cliargs lua53Packages.luacheck lua53Packages.luacov lua53Packages.luadbi lua53Packages.luadbi-mysql lua53Packages.luadbi-postgresql lua53Packages.luadbi-sqlite3 lua53Packages.luaepnf lua53Packages.luaevent lua53Packages.luaexpat lua53Packages.luaffi lua53Packages.luafilesystem lua53Packages.lualogging lua53Packages.luaossl lua53Packages.luaposix lua53Packages.luarepl lua53Packages.luarocks lua53Packages.luarocks-nix lua53Packages.luasec lua53Packages.luasocket lua53Packages.luasql-sqlite3 lua53Packages.luassert lua53Packages.luasystem lua53Packages.luautf8 lua53Packages.luazip lua53Packages.luuid lua53Packages.luv lua53Packages.lyaml lua53Packages.markdown lua53Packages.mediator_lua lua53Packages.moonscript lua53Packages.mpack lua53Packages.nvim-client lua53Packages.penlight lua53Packages.plenary-nvim lua53Packages.rapidjson lua53Packages.readline lua53Packages.say lua53Packages.std-_debug lua53Packages.std-normalize lua53Packages.stdlib lua53Packages.vstruct luajitPackages.alt-getopt luajitPackages.argparse luajitPackages.basexx luajitPackages.binaryheap luajitPackages.bit32 luajitPackages.busted luajitPackages.cassowary luajitPackages.cjson luajitPackages.compat53 luajitPackages.cosmo luajitPackages.coxpcall luajitPackages.cqueues luajitPackages.cyrussasl luajitPackages.dkjson luajitPackages.fifo luajitPackages.gitsigns-nvim luajitPackages.http luajitPackages.inspect luajitPackages.ldbus luajitPackages.ldoc luajitPackages.lgi luajitPackages.linenoise luajitPackages.ljsyscall luajitPackages.lpeg luajitPackages.lpeg_patterns luajitPackages.lpeglabel luajitPackages.lpty luajitPackages.lrexlib-gnu luajitPackages.lrexlib-pcre luajitPackages.lrexlib-posix luajitPackages.lua-cmsgpack luajitPackages.lua-iconv luajitPackages.lua-lsp luajitPackages.lua-messagepack luajitPackages.lua-resty-http luajitPackages.lua-resty-jwt luajitPackages.lua-resty-openidc luajitPackages.lua-resty-openssl luajitPackages.lua-resty-session luajitPackages.lua-term luajitPackages.lua-toml luajitPackages.lua-yajl luajitPackages.lua-zlib luajitPackages.lua_cliargs luajitPackages.luabitop luajitPackages.luacheck luajitPackages.luacov luajitPackages.luadbi luajitPackages.luadbi-mysql luajitPackages.luadbi-postgresql luajitPackages.luadbi-sqlite3 luajitPackages.luaepnf luajitPackages.luaevent luajitPackages.luaexpat luajitPackages.luafilesystem luajitPackages.lualogging luajitPackages.luaossl luajitPackages.luaposix luajitPackages.luarepl luajitPackages.luarocks luajitPackages.luarocks-nix luajitPackages.luasec luajitPackages.luasocket luajitPackages.luasql-sqlite3 luajitPackages.luassert luajitPackages.luasystem luajitPackages.luautf8 luajitPackages.luazip luajitPackages.luuid luajitPackages.luv luajitPackages.lyaml luajitPackages.markdown luajitPackages.mediator_lua luajitPackages.moonscript luajitPackages.mpack luajitPackages.nvim-client luajitPackages.penlight luajitPackages.plenary-nvim luajitPackages.rapidjson luajitPackages.readline luajitPackages.say luajitPackages.std-_debug luajitPackages.std-normalize luajitPackages.stdlib luajitPackages.vstruct luakit minitube mnemosyne mpc-qt mpv mpv-unwrapped mpvScripts.mpris mudlet neovim-qt neovim-qt-unwrapped neovim-remote neovim-unwrapped nvimpager openrussian-cli plex-media-player plex-mpv-shim prosody python310Packages.mpv python39Packages.mpv qimgv quvi sile somafm-cli stremio sublime-music telegram-cli vimPlugins.minimap-vim vis weechatScripts.weechat-matrix-bridge wordgrinder ytfzf z-lua
```
looks good. Some of the errors come from packages using the wrong lua version I guess

**Comment by teto** [comment]
seems to pass CI on darwin, merging.

**Pull request merged**